### PR TITLE
Add initial client implementation

### PR DESCRIPTION
**Pull request by bgins**
## Summary

This PR implements a client that conforms to the initial spec. It also adds a last modified timestamp (`mtime`) to each track that is set by the client. The client uses this data to avoid overwriting local data with stale server data.

This PR implements the following features:

- An initial JavaScript clock implementation that corrects for drift by comparing with web audio currentTime
- A note grid for two users with four notes
- Start and stop transport controls
- A transport display that shows the current note when playing
- A dropdown to select the current user (the user is `A` at initialization)
- A websocket implementation to send local data and receive server data

## Testing

In the `web` directory, run `npm install` and then `npm start`. Open a browser window to `localhost:1234` and click the `Start` button. Check the browser console to see that things are running, notes are changing and the echo server is pinging back with updates. Change the selected user and play around with their note grid.

## Additional context

The websocket is hardcoded to connect to `wss://echo.websocket.org` for sanity checking, but we should update this to our server endpoint.

The browser console logs the current note, local data immediately before it sent to the server, and server data whenever data is received from the server. This can be useful for seeing the sequence of events and how local data is updated with server data.

### CODE REVIEW

File: web/src/websocket.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
+export class WebSocketConnection {
+  websocket: WebSocket;
+  isAlive: boolean = false;
+  messageHandler: (notes: Notes) => void;
+
+  constructor(messageHandler) {
+    this.websocket = new WebSocket('wss://echo.websocket.org');
```
Comment (bgins) [inline web/src/websocket.ts:7]: We will want to update this to use our server's endpoint.

File: web/src/.env [hunk 2]
```diff
@@ -0,0 +1 @@
+CURRENT_USER='A'
```
Comment (bgins) [inline web/src/.env:1]: This environment variable sets the default user at app initialization.

File: web/src/global.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+type Session = {
+  A: Track
+  B: Track
+}
```
Comment (bgins) [inline web/src/global.d.ts:4]: Local data and server data should be represented as a `Session`.

File: web/src/.env (hunk 2 above)
Comment (codyshepherd) [reply web/src/.env:1]: So this is just effectively a shim for the POC correct? This isn't how we would do this when the app was functionally deployed?

**Review by codyshepherd — APPROVED** [review]
LGTM. I haven't tried running it locally yet, though

File: web/src/.env (hunk 2 above)
Comment (bgins) [reply web/src/.env:1]: Yep, exactly. In the real app, we will want auth and sign in page. This and the dropdown to select the current user are just a convenience for the POC.

**Pull request merged**